packages/builder/src/components/base/formValidation.ts

```diff
@@ -17,40 +20,70 @@ export async function validateProjectForm(inputs: FormInputs) {
 }
 
 export async function validateApplication(
-  defaultInputs: RoundApplicationQuestion[],
-  formInputs: DynamicFormInputs
+  questions: RoundApplicationQuestion[],
+  answers: RoundApplicationAnswers
 ) {
-  const schema = defaultInputs.reduce((acc, input) => {
-    const { id, required, type } = input;
-    if (type === "RECIPIENT") {
+  const schema = questions.reduce((acc, input) => {
+    let validation;
+
+    if (input.type === "project") {
+      return acc;
+    }
+
+    if (input.type === "email") {
+      validation = string().email(
+        `${input.title} must be a valid email address`
+      );
+
+      if (input.required) {
+        validation = validation.required(`${input.title} is required`);
+      }
+    } else if (input.type === "address") {
+      validation = string().matches(/^0x[a-fA-F0-9]{40}$/g, {
```

Comment:
@boudra maybe we can use `ethers.isAddress` to check the eip55 mixed-case checksum